### PR TITLE
fix: [ErrorException] Undefined array key in `spark phpini:check`

### LINKED ISSUE #8803: Bug: [ErrorException] Undefined array key "opcache.enable"
### PHP Version

8.3

### CodeIgniter4 Version

4.5.1

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

--

### What happened?

```console
P:\C4.5.1>php spark phpini:check

CodeIgniter v4.5.1 Command Line Tool - Server Time: 2024-04-18 17:06:41 UTC+00:00

[ErrorException]
Undefined array key "opcache.enable"
at SYSTEMPATH\Security\CheckPhpIni.php:155

Backtrace:
  1    SYSTEMPATH\Security\CheckPhpIni.php:155
       CodeIgniter\Debug\Exceptions()->errorHandler(2, 'Undefined array key "opcache.enable"', 'P:\\laragon\\www\\C4.5.1\\system\\Security\\CheckPhpIni.php', 155)

  2    SYSTEMPATH\Security\CheckPhpIni.php:34
       CodeIgniter\Security\CheckPhpIni::checkIni()

  3    SYSTEMPATH\Commands\Utilities\PhpIniCheck.php:73
       CodeIgniter\Security\CheckPhpIni::run()

  4    SYSTEMPATH\CLI\Commands.php:70
       CodeIgniter\Commands\Utilities\PhpIniCheck()->run([])

  5    SYSTEMPATH\CLI\Console.php:48
       CodeIgniter\CLI\Commands()->run('phpini:check', [])

  6    SYSTEMPATH\Boot.php:338
       CodeIgniter\CLI\Console()->run()

  7    SYSTEMPATH\Boot.php:104
       CodeIgniter\Boot::runCommand(Object(CodeIgniter\CLI\Console))

  8    ROOTPATH\spark:84
       CodeIgniter\Boot::bootSpark(Object(Config\Paths))
```

### Steps to Reproduce

**opcache** is not required for codeigniter. It is disabled by default in **php.ini** (`;zend_extension=opcache`).
This means that there are no `ini_get_all() ` by **opcache**  configuration options.

### Expected Output

If `;zend_extension=opcache` is set in file **php.ini**, the output of the command is as follows:
```console
P:\C4.5.1> php spark phpini:check

CodeIgniter v4.5.1 Command Line Tool - Server Time: 2024-04-18 17:15:45 UTC+00:00

+------------------------+-------------------------------+-------------------------------+-------------+-----------------------+
| Directive              | Global                        | Current                       | Recommended | Remark                |
+------------------------+-------------------------------+-------------------------------+-------------+-----------------------+
| error_reporting        | 32767                         | 32767                         | 5111        |                       |
| display_errors         | 1                             | 1                             | 0           |                       |
| display_startup_errors | 1                             | 1                             | 0           |                       |
| log_errors             | 1                             | 1                             |             |                       |
| error_log              | P:/laragon/tmp/php_errors.log | P:/laragon/tmp/php_errors.log |             |                       |
| default_charset        | UTF-8                         | UTF-8                         | UTF-8       |                       |
| memory_limit           | 512M                          | 512M                          |             | > post_max_size       |
| post_max_size          | 2G                            | 2G                            |             | > upload_max_filesize |
| upload_max_filesize    | 2G                            | 2G                            |             | < post_max_size       |
| request_order          | GP                            | GP                            | GP          |                       |
| variables_order        | GPCS                          | GPCS                          | GPCS        |                       |
| date.timezone          | UTC                           | UTC                           | UTC         |                       |
| mbstring.language      | neutral                       | neutral                       | neutral     |                       |
+------------------------+-------------------------------+-------------------------------+-------------+-----------------------+

```

### Anything else?

**solution:**

```php 
   /**
     * @internal Used for testing purposes only.
     * @testTag
     */
    public static function checkIni(): array
    {
        
        $items = [
            'error_reporting'         => ['recommended' => '5111'],
            'display_errors'          => ['recommended' => '0'],
            'display_startup_errors'  => ['recommended' => '0'],
            'log_errors'              => [],
            'error_log'               => [],
            'default_charset'         => ['recommended' => 'UTF-8'],
            'memory_limit'            => ['remark' => '> post_max_size'],
            'post_max_size'           => ['remark' => '> upload_max_filesize'],
            'upload_max_filesize'     => ['remark' => '< post_max_size'],
            'request_order'           => ['recommended' => 'GP'],
            'variables_order'         => ['recommended' => 'GPCS'],
            'date.timezone'           => ['recommended' => 'UTC'],
            'mbstring.language'       => ['recommended' => 'neutral']

        ];

        $ini    = ini_get_all();

        if (array_key_exists('opcache.enable', $ini)) {
            $itemsOpcache = [
                'opcache.enable'          => ['recommended' => '1'],
                'opcache.enable_cli'      => [],
                'opcache.jit'             => [],
                'opcache.jit_buffer_size' => [],
            ];
            
            $items = array_merge($items, $itemsOpcache);
        }

        $output = [];
        
        foreach ($items as $key => $values) {
            $output[$key] = [
                'global'      => $ini[$key]['global_value'],
                'current'     => $ini[$key]['local_value'],
                'recommended' => $values['recommended'] ?? '',
                'remark'      => $values['remark'] ?? '',
            ];
        }

        // [directive => [current_value, recommended_value]]
        return $output;
    }
```

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes #8803

```console
$ ./spark phpini:check

CodeIgniter v4.5.1 Command Line Tool - Server Time: 2024-04-19 00:01:44 UTC+00:00

+-------------------------+----------+----------+-------------+-----------------------+
| Directive               | Global   | Current  | Recommended | Remark                |
+-------------------------+----------+----------+-------------+-----------------------+
| error_reporting         | 32767    | 32767    | 5111        |                       |
| display_errors          | 1        | 1        | 0           |                       |
| display_startup_errors  | 1        | 1        | 0           |                       |
| log_errors              | 1        | 1        |             |                       |
| error_log               |          |          |             |                       |
| default_charset         | UTF-8    | UTF-8    | UTF-8       |                       |
| memory_limit            | 1G       | 1G       |             | > post_max_size       |
| post_max_size           | 8M       | 8M       |             | > upload_max_filesize |
| upload_max_filesize     | 2M       | 2M       |             | < post_max_size       |
| request_order           | GP       | GP       | GP          |                       |
| variables_order         | GPCS     | GPCS     | GPCS        |                       |
| date.timezone           | UTC      | UTC      | UTC         |                       |
| mbstring.language       | neutral  | neutral  | neutral     |                       |
| opcache.enable          | disabled | disabled | 1           |                       |
| opcache.enable_cli      | disabled | disabled |             |                       |
| opcache.jit             | disabled | disabled |             |                       |
| opcache.jit_buffer_size | disabled | disabled |             |                       |
+-------------------------+----------+----------+-------------+-----------------------+
```

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by ddevsr — APPROVED** [review]

File: system/Security/CheckPhpIni.php [hunk 1]
```diff
@@ -142,8 +142,8 @@ public static function checkIni(): array
 
         foreach ($items as $key => $values) {
             $output[$key] = [
-                'global'      => $ini[$key]['global_value'],
-                'current'     => $ini[$key]['local_value'],
+                'global'      => array_key_exists($key, $ini) ? $ini[$key]['global_value'] : 'disabled',
```
Comment (samsonasik) [inline system/Security/CheckPhpIni.php:145]: array_key_exists same condition can be moved to before `$output[$key]` definition, eg:

```php
$hasKeyInIni = array_key_exists($key, $ini);
$output[$key] = [
```

so no need double call.

Comment (kenjis) [reply system/Security/CheckPhpIni.php]: Exactly. Done.

**Review by samsonasik — APPROVED** [review]

**Comment by datamweb** [comment]
Screenshot 2024-04-19 152847

Colors should not be red for **opcache.enable_cli, opcache.jit, opcache.jit_buffer_size** ?

**Comment by kenjis** [comment]
The PHP ini default for `opcache.enable_cli` is `0`.
https://www.php.net/manual/en/opcache.configuration.php
Also, JIT is disabled by default, becuase `opcache.jit_buffer_size` is `0`.

Since what to do with these settings depends on the situation, 
I have not defined recommended values.

If enabling them improves performance, they should be enabled。
If they decrease performance, they should remain disabled.

If we set the recommended values, we need to prove that it improves performance in most use cases.

**Pull request merged**